File: showcase/js/examples/fizzbuzz.js
// The classic, with a twist: rules live in data, not in an if-chain.
const rules = [
  { div: 15, word: "FizzBuzz" },
  { div: 3, word: "Fizz" },
  { div: 5, word: "Buzz" },
];

for (let i = 1; i <= 20; i++) {
  const rule = rules.find(r => i % r.div === 0);
  console.log(rule ? rule.word : i);
}

File: showcase/perl/examples/fizzbuzz.pl
# The classic. Statement modifiers, string comparison, a C-style loop.
for (my $i = 1; $i <= 20; $i++) {
    my $out = '';
    $out .= 'Fizz' if $i % 3 == 0;
    $out .= 'Buzz' if $i % 5 == 0;
    $out = $i if $out eq '';
    print "$out\n";
}
